sdk/nodejs/cmd/run/run.ts

```diff
@@ -215,6 +216,28 @@ ${defaultMessage}`);
 
     programStarted();
 
+    if (isQueryModeEnabled()) {
```

Comment:
I think I'd much rather this code live in `stack.ts`, which is where we register the `Stack` resource, e.g.:

```
if (!isQueryModeEnabled()) {
   new Stack(init);
} else {
   init();
}

Reply:
The test mode stuff is already here, though. What do you think we should do about that?

Reply:
What do you mean? This file just parses arguments and then goes to `runInPulumiStack`, right? I'd expect the test mode stuff to be really deep in the runtime where we end up doing the actual RPC calls to the engine.

Reply:
I mean that this is basically identical to the code like 4 lines below -- it won't be after I fix it, though. But I'd planned to keep it all in one file.

Reply:
Also it would be a little weird to `runInPulumiStack` with no stack, no?

Reply:
From an API perspective, I think it would be a little weird to run a query without using "runInPulumiStack". In my mind, the name of that function (and the way we've used it) implies that we're running in the context of a stack, which we are even when we are querying. The only difference is that we don't need to register the root Stack and its outputs when we're querying.

I'm fine either way, but I would prefer that we rename or document `runInPulumiStack` aggressively to mean that it should only be used for updates, or we can make `runInPulumiStack` work for queries too. I'd favor the second, but I leave that up to you.

Reply:
Oh, that makes more sense. Yes let's do it your way.